R/knit_print.R

```diff
@@ -1,11 +1,19 @@
-
+paged_table_html = function(x) {
+  paste(
+    "<div data-pagedtable>",
+    "  <script data-pagedtable-source type=\"application/json\">",
+    jsonlite::toJSON(head(x, 1000)),
+    "  </script>",
+    "</div>",
+    sep = "\n"
+  )
+}
 
 knit_print.data.frame = function(x, ...) {
   context <- render_context()
   if (!is.null(context$df_print)) {
     if (identical(context$df_print, knitr::kable)) {
-      res = paste(c('<div class="kable-table">', '', knitr::kable(x), '', '</div>'),
-                  collapse = "\n")
+      res = paged_table_html(x)
```

Comment:
We can't do this here because this code runs for all rmarkdown formats (i.e. this would result in your paged_table_html running for PDFs!!!). Rather, for the `html_notebook` format we need to pass our df printing function for the `df_print` argument to `output_format` (i.e. where we currently pass the string "kable" in `html_notebook` we pass a function).